spring-integration-core/src/main/java/org/springframework/integration/router/ErrorMessageExceptionTypeRouter.java

```diff
@@ -67,7 +68,9 @@ private void populateClassNameMapping(Set<String> classNames) {
 
 	private Class<?> resolveClassFromName(String className) {
 		try {
-			return ClassUtils.forName(className, getApplicationContext().getClassLoader());
+			ClassLoader classLoader = getApplicationContext() == null ? getClass().getClassLoader()
```

Comment:
I would prefer do not fallback to the `getClass().getClassLoader()`.
The `Assert.state()` in this method should be great compromise.